postgresql/database.go

```diff
@@ -59,6 +59,10 @@ type cachedStatement struct {
 	query string
 }
 
+func (c *cachedStatement) OnPurge() {
+	c.Stmt.Close()
```

Comment:
It's hard to tell what would happen in this case, I think we'll be safe ignoring this since we won't be deleting statements that are in use, statements will be dereferenced by the cache, which I think is fine. But again it's hard to tell what will actually happen, so I'll add a test case and see.